dawn/renderer/reducers/editor.js

```diff
@@ -30,12 +30,7 @@ const editor = (state = defaultEditorState, action) => {
         latestSaveCode: action.code,
       };
     case 'CREATE_NEW_FILE':
-      return {
-        ...state,
-        filepath: null,
-        editorCode: '',
-        latestSaveCode: '',
-      };
+      return state;
```

Comment:
I changed it so that `sagas.js` intercepts the 'CREATE_NEW_FILE' action to call the openFile saga. This was so a dialog would appear if there are unsaved changes when the user tries to create a new file. In the openFile saga, after success, the action "OPEN_FILE_SUCCEEDED" is dispatched. Thus, I changed the flow so that from 'CREATE_NEW_FILE', after success, 'OPEN_FILE_SUCCEEDED' would be dispatched in place with a null filepath, empty editorCode, and empty latestSaveCode.
That being said, I'll remove this code segment as an unchanged state is not the state we necessarily desire, and this was redundant anyways for returning same state.